### PR TITLE
webgpu/shader: Validate semicolons

### PR DESCRIPTION
**Pull request by ben-clayton**
Fixes: #1685

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Review by zoddicus — APPROVED** [review]
Would testing something like `stmt;;` be worth-while? I am not sure off hand if an empty statement/double semis should be accepted or rejected

**Comment by ben-clayton** [comment]
> Would testing something like `stmt;;` be worth-while? I am not sure off hand if an empty statement/double semis should be accepted or rejected

Done, as two additional tests for function scope and compound statement.

File: src/webgpu/shader/validation/parse/semicolon.spec.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+  .fn(t => {
+    t.expectCompileResult(/* pass */ true, `enable f16;`);
+    t.expectCompileResult(/* pass */ false, `enable f16`);
+  });
+
+g.test('after_struct_decl')
+  .desc(`Test that a semicolon can be placed after an struct declaration.`)
+  .fn(t => {
+    t.expectCompileResult(/* pass */ true, `struct S { x : i32 };`);
+    t.expectCompileResult(/* pass */ true, `struct S { x : i32 }`);
```
Comment (dj2) [inline src/webgpu/shader/validation/parse/semicolon.spec.ts:22]: What about `struct S { x: i32; }` should be rejected?

**Review by dj2 — COMMENTED** [review]
The other cases I think fall into validation of other things (like `if (a;b)` is invalid, or `for(;;;;)` etc.). I think those would go in the validation of the construct.

File: src/webgpu/shader/validation/parse/semicolon.spec.ts (hunk 1 above)
Comment (ben-clayton) [reply src/webgpu/shader/validation/parse/semicolon.spec.ts:34]: Done.

**Review by dj2 — APPROVED** [review]

**Pull request merged**